### PR TITLE
Change component attribute name 'state' to 'display', Fix #318

### LINKED ISSUE #318: Rename `state` on components to `display`
Summary (short):
---
The name `state` for whether or not to display the component is ambiguous, it'd be better to rename it.

Description:
---
Discussed in #309, the name `state` for displaying or not displaying a component is a bit confusing.

Reproduction:
---
_1. I did this_
_2. Then i did this_
_3. And I expected X but got Y._

Screenshots, evidence:
---
:x:

### PR DESCRIPTION
**Pull request by 107g**
### Warframe Hub Pull Request
---
**Summary (short):**
Change the name of a local component attribute named 'state' to display.

---
**Description:**
The local states for components had an ambiguously named attribute called 'state'. It is a boolean that tells if the panel should be displayed or not. I changed it to 'display'. However there was already another attribute called 'display' that stored the name of the panel to display on the settings. So I changed that one to 'display_name'.

I noticed however that turning on/off the components takes some time (the page freezes for a second). I don't think it's caused by my changes but it might be worth looking into.

---
**Fixes issue (include link):**
#318 

---
**Mockups, screenshots, evidence:**
None, check the diffs.

---

(Check one)
- [x] Issue fix
- [ ] Suggestion fulfillment

### CODE REVIEW

**Comment by 107g** [comment]
The Travis-CI Branch check still hasn't triggered properly. For the next pull request I'll try to re-fork the repo and configure it properly before making any changes.

**Comment by TobiTenno** [comment]
yeah, i dunno what's going on with it, it's not letting me force a branch update either

could you squash your commits and cherrypick on a new branch? I don't know why it's doing this, i'm sorry.

**Comment by TobiTenno** [comment]
Or if you can try clicking the "Update Branch" prompt to get it to re-merge dev in so you get the other pr i just merged in

File: src/assets/json/components.json [hunk 1]
```diff
@@ -1,25 +1,25 @@
 {
   "event": {
-    "state": true,
-    "display": "Events",
+    "display": true,
+    "display_name": "Events",
```
Comment (TobiTenno) [inline src/assets/json/components.json]: could you make them camel instead of snake case?

**Review by TobiTenno — CHANGES_REQUESTED** [review]
Other than that, it looks good.

**Comment by 107g** [comment]
#321

**Pull request closed without merging**